Add changes to CFLAGS for BUILD_SHARED_LIBS equal OFF

kadorken opened the pull request:
Finally (I think) figured out pull requests on github. Hopefully this change is clear as to its intent to allow static linking of libssh2 with static linked openssl.

kadorken commented:
This is for issue #130

alamaison commented:
The change in this PR causes the compiler on windows to link statically against the CRT if BUILD_SHARED_LIBS is OFF. This doesn't make libssh2 link statically against OpenSSL. Or did you just mean that this is a necessary prerequisite for that to be possible?

A couple of comments:

- linking statically against the CRT should not be tied to building libssh2 as a static library. It's valid and common to build libssh2 static but still use the CRT via DLL. Instead we should introduce a new flag, LIBSSH2_WIN32_USE_STATIC_CRT, that enables the functionality you've added.

- I'm surprised such a flag doesn't already exist at the CMake level. Messing with compiler flags manually is generally a last resort in CMake. Are you sure we can't do this without?

kadorken commented:
In my usage, I was intending openssl to be built for static linking in parallel with libssh2. I was under the impression that the Cmake 'find' procedure for openssl looks for the .LIB files for static, or the .DLL for non-static. I had problems with Cmake  finding OpenSSL libraries under strange places (like Strawberry Perl !) other than the one I was trying to build statically).

I looked for a flag in CMake to control the linking, but I always ended up at [https://cmake.org/Wiki/CMake_FAQ#How_can_I_build_my_MSVC_application_with_a_static_runtime.3F](https://cmake.org/Wiki/CMake_FAQ#How_can_I_build_my_MSVC_application_with_a_static_runtime.3F)

I don't object to having a separate flag for the linking change; I didn't want to introduce such a flag since I'm not familiar with the overall direction of libssh2. It is a simple change

kadorken commented:
I think I managed to make the change and added to the Pull request the separate option LIBSSH2_WIN32_USE_STATIC_CRT.

Added it to the docs/INSTALL_CMAKE file as well.

simon-p-r commented:
Can this PR please be merged?

illera88 commented:
Eooooooo can this PR get merged?
